### PR TITLE
Config setting feature

### PR DESCRIPTION
**Pull request by laub1199**
Added and `!config` message handler, allowing users to set DallE image size with `!config dalle size 512x512`.
The message format will be in `!config <target> <type> <value>`.
Currently only added `dalle` with type `size` and values of `256x256` `512x512` and `1024x1024` according to the OpenAI Docs.
Also saved some rooms for more config for further development.

### CODE REVIEW

**Comment by andrewssobral** [comment]
I would like to propose `dalle-config` instead of `config`,  we may have the same for gpt or others ai assistants.

**Comment by laub1199** [comment]
> I would like to propose `dalle-config` instead of `config`, we may have the same for gpt or others ai assistants.

Sure thing! Lemme refactor it a bit

**Review by Zain-ul-din — CHANGES_REQUESTED** [review]
choose a better name for `config.ts` may be in the future we'll have a config file for the whole app. It can cause ambiguity between `config.ts` and real `config`

**Comment by laub1199** [comment]
used `aiConfig` and `dalle-config` instead of just `config`

File: src/index.ts [hunk 1]
```diff
@@ -8,12 +8,14 @@ dotenv.config();
...
 import { handleMessageGPT } from './gpt'
 import { handleMessageDALLE } from './dalle'
+import { handleMessageCONFIG } from "./config/dalle-config";
 
 // Prefixes
 const prefixEnabled = process.env.PREFIX_ENABLED == "true"
 const shouldReplySelf = process.env.REPLY_SELF_ENABLED == "true"
 const gptPrefix = '!gpt'
 const dallePrefix = '!dalle'
+const configPrefix = '!dalle'
```
Comment (andrewssobral) [inline src/index.ts:18]: There's an issue here no?

Comment (Zain-ul-din) [reply src/index.ts]: @andrewssobral, please sync your branch with `master`.

**Review by Zain-ul-din — COMMENTED** [review]
looks good

**Comment by laub1199** [comment]
Have merged the updated `master` to this PR

File: src/ai-config.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { aiConfigTarget, aiConfigTypes, aiConfigValues } from "./types/ai-config";
+
+const aiConfig = {
+    dalle: {
+        size: "512x512",
+    },
+    // chatgpt: {}
+}
+
+const handleMessageAICONFIG = async (message: any, prompt: any) => {
```
Comment (navopw) [inline src/ai-config.ts:11]: handleMessageAIConfig

**Comment by navopw** [comment]
I didn't understand why we dont use `!config` ❓

File: src/ai-config.ts (hunk 2 above)
Comment (laub1199) [reply src/ai-config.ts:11]: This was due to the `handleMessageDALLE` and `handleMessageGPT`🤔

**Comment by laub1199** [comment]
> I didn't understand why we dont use `!config` ❓ 

Both `!config` and `!aiconfig' sound good to me, just see which one will be more suitable for you guys.

**Comment by navopw** [comment]
Sorry, I restructured the code a bit.

You can add your options into the config.ts (especially the DALLE resolution)

Please adjust it and I will merge :)

Thanks for contributing! ❤️

**Comment by laub1199** [comment]
Sure, will add the `aiConfig` into the `config.ts` in a bit and rename the handler to `handleMessageAIConfig`.
What abt `!aiconfig`, are we going for this or just `!config`.

**Comment by laub1199** [comment]
`!aiconfig` had been modified to `!config`
`handleMessageAICONFIG` has been modified to `handleMessageAIConfig`
Dalle related config types had been moved to `types/dalle-config.ts` and `types/ai-config.ts`
Config Message Handler had been moved to `handlers/ai-config.ts`
The actual config is now in `handlers/ai-config.ts` while the dalle or other ai assistants' config can be store in the `aiConfig` object

**Comment by navopw** [comment]
This code actually breaks the dalle...

This is the data we send to dalle, size should be 1024x1024 and not "large"

'{"prompt":"big brain","n":1,"size":"large","response_format":"b64_json"}',

We have to fix this asap.

**Comment by navopw** [comment]
I think we need to clean this up a bit more,

this enum looks weird :D

**Comment by navopw** [comment]
But works now!

Thank you! <3

**Comment by laub1199** [comment]
glad to contribute! will keep track of this repo, hopefully, more contributions in the future:D

**Pull request merged**